packages/ra-core/src/auth/Authenticated.spec.tsx

```diff
@@ -32,36 +49,53 @@ describe('<Authenticated>', () => {
             checkError: jest.fn().mockResolvedValue(''),
             getPermissions: jest.fn().mockResolvedValue(''),
         };
-
+        const store = createAdminStore();
+        const dispatch = jest.spyOn(store, 'dispatch');
         const history = createMemoryHistory();
 
-        const { dispatch } = renderWithRedux(
-            <Router history={history}>
-                <AuthContext.Provider value={authProvider}>
-                    <Authenticated>
-                        <Foo />
-                    </Authenticated>
-                </AuthContext.Provider>
-            </Router>
+        const Login = () => {
+            const location = useLocation();
+
+            return (
+                <div aria-label="nextPathname">
+                    {location.state.nextPathname}
+                </div>
+            );
+        };
+        render(
+            <Provider store={store}>
+                <CoreAdminContext
+                    authProvider={authProvider}
+                    dataProvider={() => Promise.resolve()}
+                    history={history}
+                >
+                    <Routes>
+                        <Route
+                            path="/"
+                            element={
+                                <Authenticated>
+                                    <Foo />
+                                </Authenticated>
+                            }
+                        />
+                        <Route path="/login" element={<Login />} />
+                    </Routes>
+                </CoreAdminContext>
+            </Provider>
         );
         await waitFor(() => {
             expect(authProvider.checkAuth.mock.calls[0][0]).toEqual({});
             expect(authProvider.logout.mock.calls[0][0]).toEqual({});
             expect(dispatch).toHaveBeenCalledTimes(2);
             expect(dispatch.mock.calls[0][0]).toEqual(
-                showNotification('ra.auth.auth_check_error', 'warning', {
-                    messageArgs: {},
-                    undoable: false,
-                })
+                showNotification('ra.auth.auth_check_error', 'warning')
             );
             expect(dispatch.mock.calls[1][0]).toEqual({
                 type: 'RA/CLEAR_STATE',
             });
-            expect(history.location.pathname).toEqual('/login');
-            expect(history.location.state).toEqual({
-                nextPathname: '/',
-                nextSearch: '',
-            });
+            expect(screen.getByLabelText('nextPathname').innerHTML).toEqual(
```

Comment:
this test sin't readable. I understand you're testing that we are on the login page, but it's really not apparent here.

Reply:
Any suggestion?

Reply:
As discussed, I didn't change the original tests. Testing more things si out of scope for this PR